src/components/Productpage.js

```diff
@@ -0,0 +1,85 @@
+import React , {useEffect, useState} from 'react'
+import {useParams} from 'react-router-dom'
+import {useData} from '../contexts/DataContext'
+import {getRatingType , addToCartHandler} from '../util'
+import Navbar from './Navbar'
+import '../styles/productpage.css'
+import {Link} from 'react-router-dom'
+import axios from 'axios'
+const Productpage = () => {
+    const {productId} = useParams()
+    const {state : {cart , wishList} , dispatch} = useData() 
+    const addToWishListHandler = async () => {
+            
+            dispatch({type : 'ADD_TO_WISHLIST' , payload : details})
+    }
+    const isWishListed = (prodId) => {
+        return wishList.reduce((acc , curr) => {
```

Comment:
you r writing this function in multiple components ! write this in one file & import it !